OpenTrons/opentrons-api
# fix(app): clone run with RTPs from HistoricalProtocolRun

closes [RQA-2601](https://opentrons.atlassian.net/browse/RQA-2601)

# Overview

Here, I extend the useCloneRun hook to optionally create a new analysis for a protocol with RTP
override values that will be referenced at protocol setup. This provides functionality for cloning a
HistoricalProtocolRun with its RTP values preserved.


# Test Plan

1. Run an RTP protocol with any parameter set
2. Rerun that same protocol with different parameters on the same robot
3. Navigate to device details run history
4. Click the overflow menu > "Rerun protocol now"
5. Verify that the Parameters tab shows the RTP for the selected run and not the most recent run
6. Repeat 4 and 5, but clicking the overflow menu > "View protocol run record"
7. Repeat 4 and 5, but clicking the date of the run record rather than "Rerun now" 

# Changelog

- update `useCloneRun` to optionally trigger a new protocol analysis, properly invalidate queries to `/runs` and `/protocols`
- use run record rather than most recent analysis able on `Parameters` and `RunPreview`
- add and export constant for terminal run statuses
- fix `runTimeParameters` key on `Run` type
- add ProtocolRun > Parameters `InfoScreen` for run canceled before beginning

# Review requests

auth, @shlokamin per pairing

# Risk assessment

medium

[RQA-2601]: https://opentrons.atlassian.net/browse/RQA-2601?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
